loglist2/loglist2.go

```diff
@@ -136,6 +136,76 @@ type TreeHead struct {
 	TreeSize int64 `json:"tree_size"`
 }
 
+// String method returns printable name of the state.
+func (ls *LogStates) String() string {
+	names := [...]string{
+		"Pending",
+		"Qualified",
+		"Usable",
+		"Frozen",
+		"Retired",
+		"Rejected",
+		"Empty"}
+	if ls == nil {
+		return names[6]
+	}
+	if ls.Pending != nil {
+		return names[0]
+	} else if ls.Qualified != nil {
+		return names[1]
+	} else if ls.Usable != nil {
+		return names[2]
+	} else if ls.Frozen != nil {
+		return names[3]
+	} else if ls.Retired != nil {
+		return names[4]
+	} else if ls.Rejected != nil {
+		return names[5]
+	}
+	return names[6]
+}
+
+// Active picks the set-up state. Returns error if multiple states set.
+func (ls *LogStates) Active() (*LogState, *FrozenLogState, error) {
+	var active *LogState
+	var activeF *FrozenLogState
+	var activeNum int
+
+	if ls == nil {
+		return nil, nil, nil
+	}
+
+	if ls.Pending != nil {
```

Comment:
This should also be a switch block

Reply:
While I totally agree with a previous switch suggestion, this one doesn't look that convenient. Unless you suggest 'fallthrough' in every clause.
Here summation happens to detect  erroneous multi-state setting.

Reply:
I suggest splitting the summation / validation into a separate function from setting the active variable. 

Perhaps you could set the ls.XXX variable to nil in the switch, and then check if there are any other set items after the switch. ie. `err = (ls.Qualified != nil) || (ls.Usable != nil) ...`

I hope that's not too pedantic.

Reply:
I just removed correctness check.